filestream: derive scanner fingerprinting from the file identity (#52147)
elastic/beats: docs/reference/filebeat/file-identity.md

| path | Files are never moved or renamed, file names are never re-used. | Simple and fast. | The most unstable option, requires to maintain immutable file paths. |
| native (default in Filebeat < 9.0) | Stable file systems, files < 64 bytes in size, ingestion without delays. | Low CPU / memory overhead. | Might cause data duplication or data loss if the file system provides unstable `inode` or `device ID` values. No support for network shares, containers or VMs.
| inode_marker | Same as `native` but `device ID` is changing. | Same as `native` + no dependency on `device ID`. | Can still cause data duplication or data loss due to unstable `inode` values provided by the file system. Also, no support for network shares, containers or VMs. |

> | fingerprint (default in Filebeat >= 9.0) | Log files with unique content.<br><br>{applies_to}`stack: ga 9.5+` Log files of any size (enhanced fingerprint).<br><br>{applies_to}`stack: ga 8.10-9.4` Log files larger than the fingerprint size (1 KB by default). | The most stable. Support for any OS, any file system, network shares, containers and VMs.<br><br>{applies_to}`stack: ga 9.5+` Ingests files smaller than the fingerprint size without delay. | Slightly higher CPU / memory usage.<br><br>{applies_to}`stack: ga 8.10-9.4` Doesn't ingest files until they reach the fingerprint size (1 KB by default). |

Reviewer: Those docs are only for 9.x
```suggestion
| fingerprint (default in Filebeat >= 9.0) | Log files with unique content.<br><br>{applies_to}`stack: ga 9.5+` Log files of any size (enhanced fingerprint).<br><br>{applies_to}`stack: ga 9.0-9.4` Log files larger than the fingerprint size (1 KB by default). | The most stable. Support for any OS, any file system, network shares, containers and VMs.<br><br>{applies_to}`stack: ga 9.5+` Ingests files smaller than the fingerprint size without delay. | Slightly higher CPU / memory usage.<br><br>{applies_to}`stack: ga 9.0-9.4` Doesn't ingest files until they reach the fingerprint size (1 KB by default). |
```

The most stable file identity implementation at the moment. This file identity is default since Filebeat version 9.0.

This file identity uses file fingerprints produced by the scanner component of the filestream input.

::::{applies-switch}
:group: fingerprint-scanner
:::{applies-item} stack: ga 9.0+
:sync: 9.0+
Scanner fingerprinting is enabled by default, so this file identity needs no extra configuration.
:::
:::{applies-item} stack: ga 8.10-8.19
:sync: 8.10-8.19
Scanner fingerprinting is disabled by default. Set `prospector.scanner.fingerprint.enabled: true` to use this file identity.

> :::

Reviewer: Those docs are only for 9.x
```suggestion
```

The most stable file identity implementation at the moment. This file identity is default since Filebeat version 9.0.

> This file identity uses file fingerprints produced by the scanner component of the filestream input. Scanner fingerprinting is enabled automatically when this file identity is used and disabled for any other file identity. The `prospector.scanner.fingerprint.enabled` setting is deprecated and ignored; Filebeat logs a warning if its value contradicts the configured `file_identity`.

Reviewer 2: ```suggestion
This file identity uses file fingerprints produced by the scanner component of the filestream input. Scanner fingerprinting is enabled automatically when this file identity is used. It is disabled for any other file identity. The `prospector.scanner.fingerprint.enabled` setting is deprecated and ignored. Filebeat logs a warning if its value contradicts the configured `file_identity`.
```

For example, Filebeat is running a filestream input with this configuration:

::::{applies-switch}
:group: fingerprint-scanner
:::{applies-item} stack: ga 9.0+
:sync: 9.0+
```yaml
prospector.scanner.fingerprint:
  offset: 32   # e.g. the first 32 bytes are always identical in every file, we skip them
  length: 1024 # strong guarantee of uniqueness
file_identity.fingerprint: ~
```
:::
:::{applies-item} stack: ga 8.10-8.19
:sync: 8.10-8.19
```yaml
prospector.scanner.fingerprint:
  enabled: true # required: the scanner computes no fingerprints by default
  offset: 32    # e.g. the first 32 bytes are always identical in every file, we skip them
  length: 1024  # strong guarantee of uniqueness
file_identity.fingerprint: ~
```
:::

> ::::

Reviewer: No need for < 9.0 docs
```suggestion
```

> {applies_to}`stack: ga 8.10-9.4` If a new log file appears, the filestream input doesn’t register it until it grows to 1056 bytes (`offset`+`length`). Files that don't grow to that size are not ingested.

Reviewer: ```suggestion
{applies_to}`stack: ga 9.0-9.4` If a new log file appears, the filestream input doesn’t register it until it grows to 1056 bytes (`offset`+`length`). Files that don't grow to that size are not ingested.
```
